### ISSUE TITLE
Reolink E1 Zoom rebooting

### ISSUE THREAD

**Issue by roderickstoner** [fwestenberg/reolink_dev#236]
It seems that when using this integration with E1 Zoom, the camera heats up and starts rebooting. When the device is disabled from HA, the rebooting stops.

**Comment by chiefcomm** [comment]
> It seems that when using this integration with E1 Zoom, the camera heats up and starts rebooting. When the device is disabled from HA, the rebooting stops.

Hi @The-IcEDFiRE, I use E1 Zoom with this integration in Home Assistant and have no issues with overheating at all.  My Reolink E1 Zoom 5MP WiFi PTZ has been working perfectly since purchase and install 1 month ago.   I have E1 Zoom, RLC-410, RLC-420 and RLC-822A all working perfectly with Home Assistant :-)

**Comment by The-IcEDFiRE** [comment]
> > It seems that when using this integration with E1 Zoom, the camera heats up and starts rebooting. When the device is disabled from HA, the rebooting stops.
> 
> Hi @The-IcEDFiRE, I use E1 Zoom with this integration in Home Assistant and have no issues with overheating at all. My Reolink E1 Zoom 5MP WiFi PTZ has been working perfectly since purchase and install 1 month ago. I have E1 Zoom, RLC-410, RLC-420 and RLC-822A all working perfectly with Home Assistant :-)

Hmm.. what firmware is the E1 zoom running? I have six E1 Zoom cams and they all reboot at a point or another..at times once every so often, in other instances, once one starts rebooting,.it keeps

**Comment by FelipeDamasio** [comment]
I'm having the same issue here.
I noticed that when I connect the camera to HA, the image in the official app gets a lot of delay and I get poor network quality notifications.
  Without any integration with HA, the speed of the app is 4.5mb ~ 5mb, but when integrated with HA, the speed is just 3mb.
The only way I found to make it stable is to let the intermediate quality in the application and in the HA use the SUB setting instead of MAIN in the stream.
I believe it is the camera's (wifi) lack of capacity to support traffic for both applications in high resolution.

**Comment by The-IcEDFiRE** [comment]
> I'm having the same issue here.
> I noticed that when I connect the camera to HA, the image in the official app gets a lot of delay and I get poor network quality notifications.
> Without any integration with HA, the speed of the app is 4.5mb ~ 5mb, but when integrated with HA, the speed is just 3mb.
> The only way I found to make it stable is to let the intermediate quality in the application and in the HA use the SUB setting instead of MAIN in the stream.
> I believe it is the camera's (wifi) lack of capacity to support traffic for both applications in high resolution.

If I use Onvif integration it works fine albeit with some lag which is understandable.. the cameras don't reboot once.. but as soon as I enable the camera with this integration, they start rebooting at random intervals. Didn't take note of decrease in video feed.

**Comment by FelipeDamasio** [comment]
Try changing the integration configuration to Sub instead of main.
![Screenshot_2](https://user-images.githubusercontent.com/54179992/128060424-c4e6987a-12a7-42b3-b77d-007f7da42e01.png)

**Comment by FelipeDamasio** [comment]
Try changing the integration configuration to Sub instead of main.
![image](https://user-images.githubusercontent.com/54179992/128060542-a6739026-e323-4c75-be6b-74c1787c9470.png)

**Comment by The-IcEDFiRE** [comment]
> Try changing the integration configuration to Sub instead of main.
> ![image](https://user-images.githubusercontent.com/54179992/128060542-a6739026-e323-4c75-be6b-74c1787c9470.png)

Hmmm...It is giving me a malformed error...

![IMG_20210803_201236](https://user-images.githubusercontent.com/63777977/128065476-73680a4f-61b3-4d6f-bb4f-fd9fe2bd9c1b.jpg)

**Comment by FelipeDamasio** [comment]
inset a space " " in the custom thumbnail

**Comment by The-IcEDFiRE** [comment]
> inset a space " " in the custom thumbnail

That worked..thanks.. now will see if it reboots.

**Comment by The-IcEDFiRE** [comment]
> > inset a space " " in the custom thumbnail
> 
> That worked..thanks.. now will see if it reboots.

It has rebooted a number of times....

![image](https://user-images.githubusercontent.com/63777977/128132881-7373f8a3-139b-4ebf-93cd-d75f3fad5e53.png)

![image](https://user-images.githubusercontent.com/63777977/128132721-d39d125a-79b6-4bc9-ad76-cbb6f17fc7a1.png)

and now I lost it... need to go and reboot it manually :(

**Comment by FelipeDamasio** [comment]
Try removing your SD card, if you have it. I replaced my SD and apparently improved this problem.

**Comment by The-IcEDFiRE** [comment]
I will try it. But why does it reboot in the first place when using this integration and not when using ONVIF or an NVR?

**Comment by D0m3L** [comment]
@The-IcEDFiRE were you able to find a solution? I've got the same issue with rebooting E1 Zoom with HA core-2021.8.7. I tried several reolink_dev releases but with no luck so far.

**Comment by The-IcEDFiRE** [comment]
> @The-IcEDFiRE were you able to find a solution? I've got the same issue with rebooting E1 Zoom with HA core-2021.8.7. I tried several reolink_dev releases but with no luck so far.

To be honest they stopped rebooting. Reolink support kept pointing to router issues, which I have no idea why. Then the router got an update and the cameras stopped rebooting. I think it is more of a coincidence though. In the meantime, there was an update to the Integration and to HA. Plus, I also had configured them to use the ONVIF integration and disabled them from the Reolink integration. Setting the stream to Sub instead of Main didn't help much and now with Main the camera still doesn't reboot.

I think it could have also been heat-related maybe and now that summer is almost over, and not as hot as before, maybe that couple of deg was a tipping point. I still get some gaps in the HA history but those aren't actual reboots... it seems that at times HA and the cameras stop communicating or something.
![image](https://user-images.githubusercontent.com/63777977/131825483-2f4a4778-5062-4e4e-9f7e-0568fd294b53.png)

Bottom line, they seem to have stopped rebooting (for now) but have no idea why.

**Comment by D0m3L** [comment]
Lucky you... :/
The regularity of restarts is suspect - accuracy down to the second - sometimes even each hour (like reolink subscription renewal period?).
Anyway, in my case the problem started to be noticeable after the summer season, although I do not exclude that it did not occur before, I just did not use this integration so often before. 

It turns out I've to integrate my Reolink in a different way, unless camera/HA/integration got new release to be tested. I'll try to rollback camera firmware to some older version just to make sure Reolink dev does not broke something in between.

Thanks for sharing your status though. Be well!

**Comment by FabioEight** [comment]
I'm experiencing this issue as well, the camera is rebooting itself only if I use reolink_dev, configured as suggested above. If I integrate the camera via ONVIF, the camera is stable and not rebooting.

**Comment by The-IcEDFiRE** [comment]
> Lucky you... :/
> The regularity of restarts is suspect - accuracy down to the second - sometimes even each hour (like reolink subscription renewal period?).
> Anyway, in my case the problem started to be noticeable after the summer season, although I do not exclude that it did not occur before, I just did not use this integration so often before.
> 
> It turns out I've to integrate my Reolink in a different way, unless camera/HA/integration got new release to be tested. I'll try to rollback camera firmware to some older version just to make sure Reolink dev does not broke something in between.
> 
> Thanks for sharing your status though. Be well!

Hi, I tried different firmware versions... same results... I think the issue is with heating (high CPU usage perhaps). I do have one of the cameras that has a lose connection in the power socket though. So check out for those too. I am now using Frigate (needed person detection) and RTMP streams (RTSP was horrible). If you just need to view the cams, ONVIF is more than enough. For those that don't have ONVIF, you can try using NEOLINK which is a bridge between the native Reolink protocol and RTSP.

**Comment by ricarva** [comment]
> Lucky you... :/ The regularity of restarts is suspect - accuracy down to the second - sometimes even each hour (like reolink subscription renewal period?). Anyway, in my case the problem started to be noticeable after the summer season, although I do not exclude that it did not occur before, I just did not use this integration so often before.
> 
> It turns out I've to integrate my Reolink in a different way, unless camera/HA/integration got new release to be tested. I'll try to rollback camera firmware to some older version just to make sure Reolink dev does not broke something in between.
> 
> Thanks for sharing your status though. Be well!

@D0m3L 

Did you have any luck in rolling back firmware and fixing the disconnects?

**Comment by D0m3L** [comment]
> 
> @D0m3L 
> 
> Did you have any luck in rolling back firmware and fixing the disconnects?
> 

Unfortunately not. I've switched my integration to shell scripts performing simple motion on/off and ptz presets at certain condition and since then I forgot about reolink_dev integration.

**Comment by ricarva** [comment]
> > @D0m3L
> > Did you have any luck in rolling back firmware and fixing the disconnects?
> 
> @ricarva Unfortunately not. I've switched my integration to shell scripts performing simple motion on/off and ptz presets at certain condition and since then I forgot about reolink_dev integration.

@D0m3L 

Thanks for the update!

I've just turned on debug logging for this integration to try and get more information on what's going on: as per your comment above, the hourly disconnects in the integration are pretty supicious.

@fwestenberg and @cpainchaud, any idea of what could be at the root of this? I'll update information from my logs as it becomes available.

**Comment by markalroberts** [comment]
Getting this problem too - had Reolink E1 Zoom working stand-alone for a few months without any problems, the attached HA and boom, it restarts every hour or so - I was on an old firmware but upgrading to latest doesn't help sadly. No SD card installed.

If anyone finds an answer to this I'd be grateful!

**Comment by The-IcEDFiRE** [comment]
Ever since, my cameras reboot even when not connected to HA... at a less
frequency... I noticed they do it mostly when at night with IR on.

**Comment by FelipeDamasio** [comment]
I've come to the conclusion that it really is a hardware problem. I bought another Reolink E1 outdoor and had no problems.
I opened my E1 zoom and realized that it doesn't have a heatsink, and that it gets too hot. Here in reverse it works fine, but in the summer, especially with the IR on, it burns my finger due to the high temperature and restarts frequently.

**Comment by markalroberts** [comment]
Tried disabling all entities other than camera (inc motion detection) but still kept resetting. Disabled all 9 entities and it’s not reset since. So I guess something to do with the continual polling of the upnp interface for the image is making this device either over heat or crash in some other way. Sadly this means that for me at least the Reolink I a not a viable HA device right now.

**Comment by l4ci** [comment]
Having the same problem with two e1 zoom cameras. Even replaced them both to sort out a device error. Wifi Signal can not be a problem. I will disabled some settings and keep the other one as is, to find the blamable one.

**Comment by DracoC77** [comment]
I've been having a similar issue with my E1 Zoom camera randomly restarting and returning to a different point than I had it before so I started digging through the HA logs but noticed that the camera restarts didn't seem consistent with the dropouts I saw in HA from the reolink_dev component. Furthermore the reolink_dev dropouts seemed to be occuring at very 1, 2 or 3 hour regular intervals (down to the second) which seems to suggest more of a hardware problem than a software problem

To further debug I added ping sensors pointed at the actual reolink cameras and confirmed that when the cameras reset (manually via power cycle, reboot via reolink app, or randomly) they do trigger downtime on those ping sensors.

Here is a plot comparing two reolink_dev sensors (the camera feed + also IR on/off) and their gaps vs. the ping sensor (I rebooted the camera once or twice in there manually for debugging).

![reolink_ping_vs_integration](https://user-images.githubusercontent.com/69522449/180257395-616b37df-b388-4227-9fea-c2b295ace946.png)

This seems to suggest that yes the E1 zoom camera and my E1 outdoor camera are sometime rebooting, but there is also some sort of periodic recurring dropout problem with the reolink_dev integration that causes dropouts at fixed intervals. I have ancdeotal evidence that this dropout behavior gets better after an HA restart (full docker down/up as I'm using container version) and ancedotally I don't see a correlation with the actual camera reboots vs. integration attached/detached (though I have not tested this part enough/thoroughly)

Is the periodic rebooting something I should start a separate thread/but for?

**Comment by ricarva** [comment]
@DracoC77 

I haven't had a reboot ever since Reolink exchanged my E1 Zoom for a new one; also, the replacement camera is a newer hardware revision.

The integration dropouts still show up in the HA graphs, but the reboots seemed to be hardware-related.

**Comment by DracoC77** [comment]
@ricarva 

What HW revision is your newer E1 Zoom camera? I'm HW IPC_523SD8 on Firmware v3.0.0.748_21120806.

**Comment by ricarva** [comment]
@DracoC77 my E1 Zoom is HW version IPC_515BSD6, older one was IPC_515SD6. Seeing that you seem to have the same HW revision, maybe it is just somekind of hardware failure.

**Comment by dalet11** [comment]
Have the same issue with 2 e1 zoom cameras.

**Comment by DracoC77** [comment]
Just sharing where I ended up on this issue:
- The HA reolink integration seems to have periodic dropout that are specific to the integration itself (and occurs on periodic intervals, either at 1, 2, 3 hour intervals triggering at specific times down to hte second, these do NOT seem correlated to the actual Reolink camera losing connection
- The Reolink cameras DO occasionally lose wifi connection for some reason, I've noticed it is most likely correlated to either too many streaming connections or may be due to general wifi dropouts
- E1 outdoor camera seems oddly better performing than the E1 zoom, may just be an artifact of my own placement
- When these cameras lose wifi they will reset their position and will NOT return to exactly the same position (due to some sort of encoder error/repetability issue)

At the end of the way I returned my E1 zoom since it was dropping link so often, but have kept my E1 outdoor since that seems to be performing better. Hope this helps folks that find this thread.
